controller/array_action/svc_cli_result_reader.py

```diff
@@ -0,0 +1,107 @@
+from controller.common.csi_logger import get_stdout_logger
+import controller.array_action.errors as controller_errors
+
+logger = get_stdout_logger()
+
+ID_PARAM_NAME = "id"
+
+
+class SVCListResultsReader:
+    """
+    Iterable object used to read raw command response from SVC array.
+    Each object in response is translated to SVCListResultsElement.
+    Input is received as string which represents '\n'-separated list of returned lines from output.
+    Line with param 'id' (e.g. 'id 1') is recognized as first line of object ans used as separator between objects
+    (e.g. in input "id 1\nname n3\nid 2\nname n2" first object starts with line 'id 1' and ends with 'id 2')
+    """
+    def __init__(self, hosts_raw_list_as_string):
+        self._hosts_raw_list = hosts_raw_list_as_string.splitlines()
+        self._current_index = 0
+        self._next_object_id = None
+        self._init_first_object_id()
+
+    def _init_first_object_id(self):
+        """
+        Set _next_object_id to id of the first object and _current_index to point to next line
+
+        Raises:
+            InvalidCliResponseError
+        """
+        while not self._next_object_id and self._current_index < len(self._hosts_raw_list):
+            line = self._hosts_raw_list[self._current_index].strip()
+            self._current_index += 1
+            if line:
+                param_name, _, param_value = line.partition(' ')
+                param_value = param_value.strip()
+                if param_name == ID_PARAM_NAME:
+                    self._next_object_id = param_value
+                else:
+                    raise controller_errors.InvalidCliResponseError(
+                        "First element is {0}. Expected {1}".format(line, ID_PARAM_NAME))
```

Comment:
shouldn't it be `First param is ...`? I thought "element" is the complete object, e.g. cli host

Reply:
fixed

Reply:
I meant that the "element" should be replaced by "param", and not remove "param"

Reply:
updated